PWGHF/D2H/Tasks/taskCharmPolarisation.cxx

```diff
@@ -874,26 +941,124 @@ struct TaskPolarisationCharmHadrons {
+
+          /// check if daughter pairs have the same mother
+          /// it should be enough to check the 1st one only (only particles from partonic events, or interactions with material, can have more than 1 mother)
+          int pdgMotherProng0 = -1;
+          int pdgMotherProng1 = -1;
+          int pdgMotherProng2 = -1;
+          if (idMothersProng0.size() > 0 && idMothersProng1.size() > 0 && idMothersProng0.at(0) == idMothersProng1.at(0)) {
+            auto mother = particles.rawIteratorAt(idMothersProng0.at(0) - particles.offset());
+            pdgMotherProng0 = std::abs(mother.pdgCode()); // PDG code of the mother
+            pdgMotherProng1 = pdgMotherProng0;
+          }
+          if (idMothersProng1.size() > 0 && idMothersProng2.size() > 0 && idMothersProng1.at(0) == idMothersProng2.at(0)) {
+            auto mother = particles.rawIteratorAt(idMothersProng1.at(0) - particles.offset());
+            pdgMotherProng1 = std::abs(mother.pdgCode()); // PDG code of the mother
+            pdgMotherProng2 = pdgMotherProng1;
+          }
+          if (idMothersProng0.size() > 0 && idMothersProng2.size() > 0 && idMothersProng0.at(0) == idMothersProng2.at(0)) {
+            auto mother = particles.rawIteratorAt(idMothersProng0.at(0) - particles.offset());
+            pdgMotherProng0 = std::abs(mother.pdgCode()); // PDG code of the mother
+            pdgMotherProng2 = pdgMotherProng0;
+          }
+
+          /// calculate inv. masses for pairs, depending on mass hypothesis
+          std::array<float, 3> pVecPion = {};
+          std::array<float, 3> pVecKaon = candidate.pVectorProng1();
+          std::array<float, 3> pVecProton = {};
+          if (iMass == charm_polarisation::MassHyposLcToPKPi::PKPi && candidate.isSelLcToPKPi() >= selectionFlagLcToPKPi) {
+            pVecProton = candidate.pVectorProng0();
+            pVecPion = candidate.pVectorProng2();
+          } else if (iMass == charm_polarisation::MassHyposLcToPKPi::PiKP && candidate.isSelLcToPiKP() >= selectionFlagLcToPKPi) {
+            pVecProton = candidate.pVectorProng2();
+            pVecPion = candidate.pVectorProng0();
+          }
+          const float massKPi = RecoDecay::m(std::array{pVecKaon, pVecPion}, std::array{o2::constants::physics::MassKPlus, o2::constants::physics::MassPiPlus});
+          const float massKProton = RecoDecay::m(std::array{pVecKaon, pVecProton}, std::array{o2::constants::physics::MassKPlus, o2::constants::physics::MassProton});
+          const float massPiProton = RecoDecay::m(std::array{pVecPion, pVecProton}, std::array{o2::constants::physics::MassPiPlus, o2::constants::physics::MassProton});
+
+          /// Fill the table for selected candidates
+          /// No need to check explicitly if candidates are selected, since the Filter is applied
+          rowCandLcBkg(invMassCharmHadForSparse, ptCharmHad, rapidity,
+                       cosThetaStarHelicity, cosThetaStarProduction, cosThetaStarBeam, cosThetaStarRandom,
```

Comment:
Do we want all the cost* axes, or maybe only one is sufficient (helicity, or configurable)?

Reply:
good point, I made it configurable just for flexibility, but indeed we are more interested on helicity. Fixed it.